src/test/java/org/neo4j/junit/jupiter/causal_cluster/PauseUnpauseServersTest.java

```diff
@@ -110,11 +111,30 @@ void pauseAllTest(int pauseMilliseconds) throws Exception {
 
 		// when
 		Set<Neo4jServer> unpaused = cluster.unpauseServers(pausedServers);
-		cluster.waitForBoltOnAll(pausedServers, Duration.ofMinutes(3));
+		cluster.waitForBoltOnAll(unpaused, Duration.ofMinutes(1));
 
 		// then
 		assertThat(unpaused).containsExactlyInAnyOrderElementsOf(pausedServers);
+		verifyClusterResumedSuccessfully();
+
+	}
+
+	private void verifyClusterResumedSuccessfully() throws TimeoutException {
```

Comment:
Minor:
We can try to write this function without code duplication 
```
private void verifyClusterResumedSuccessfully( boolean retryOnce ) throws TimeoutException
	{
		// handling recovery from a pause is a bit strange because _immediately_ after the unpause everything functions
		// as it did before. But then (usually within seconds) the cores realise that various things have timed out and
		// the cores may then be unusable for a short period while they figure out what's going on.

		// first wait for full connectivity
		DriverUtils.verifyEventuallyAllServersHaveConnectivity( cluster, Duration.ofMinutes( 5 ) );
		try
		{
			// that connectivity persists uninterrupted for 1 minute
			DriverUtils.verifyContinuouslyAllServersHaveConnectivity( cluster, Duration.ofMinutes( 2 ) );
		}
		catch ( Exception e )
		{

			if ( retryOnce )
			{
			        // if something goes wrong we retry once
			        log.warn( () -> "Connectivity lost (retrying once). Cause: " + e.toString() );
				verifyClusterResumedSuccessfully( false );
			}
		}
	}
```